src/System.Drawing.Common/tests/ImageTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using Xunit;
+
+namespace System.Drawing.Tests
+{
+    public class ImageTests
+    {
+        public static IEnumerable<object[]> InvalidBytes_TestData()
+        {
+            // IconTests.Ctor_InvalidBytesInStream_TestData an array of 2 objects, but this test only uses the
+            // 1st object.
+            foreach (object[] data in IconTests.Ctor_InvalidBytesInStream_TestData())
+            {
+                yield return new object[] { data[0] };
+            }
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [MemberData(nameof(InvalidBytes_TestData))]
+        public void FromFile_InvalidBytes_ThrowsOutOfMemoryException(byte[] bytes)
+        {
+            using (var file = TempFile.Create(bytes))
+            {
+                Assert.Throws<OutOfMemoryException>(() => Image.FromFile(file.Path));
+                Assert.Throws<OutOfMemoryException>(() => Image.FromFile(file.Path, useEmbeddedColorManagement: true));
+            }
+        }
+
+        [Fact]
+        public void FromFile_NullFileName_ThrowsArgumentNullException()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("path", () => Image.FromFile(null));
+            AssertExtensions.Throws<ArgumentNullException>("path", () => Image.FromFile(null, useEmbeddedColorManagement: true));
+        }
+
+        [Fact]
+        public void FromFile_EmptyFileName_ThrowsArgumentNullException()
+        {
+            AssertExtensions.Throws<ArgumentException>("path", null, () => Image.FromFile(string.Empty));
+            AssertExtensions.Throws<ArgumentException>("path", null, () => Image.FromFile(string.Empty, useEmbeddedColorManagement: true));
+        }
+
+        [Fact]
+        public void FromFile_LongFile_ThrowsPathTooLongException()
+        {
+            string fileName = new string('a', 261);
+
+            Assert.Throws<PathTooLongException>(() => Image.FromFile(fileName));
+            Assert.Throws<PathTooLongException>(() => Image.FromFile(fileName, useEmbeddedColorManagement: true));
+        }
+
+        [Fact]
+        public void FromFile_NoSuchFile_ThrowsFileNotFoundException()
+        {
+            Assert.Throws<FileNotFoundException>(() => Image.FromFile("NoSuchFile"));
+            Assert.Throws<FileNotFoundException>(() => Image.FromFile("NoSuchFile", useEmbeddedColorManagement: true));
```

Comment:
In theory such a file could exist.  Probably better to use something like Guid.NewGuid().ToString("N").